### PR TITLE
Support Python 3.12

### LINKED ISSUE #744: Support Python 3.12
### Problem Description
Python 3.12 was released on Oct 2, 2023.

### Expected behavior
In order to upgrade the RDT, we'll need to ensure that our external dependencies are compatible with 3.12 (pandas, Faker, etc.)

This is a blocking issue for upgrading the main SDV library to 3.12. See [SDV Issue 1704](https://github.com/sdv-dev/SDV/issues/1704) and [CTGAN Issue 324](https://github.com/sdv-dev/CTGAN/issues/324).

### PR DESCRIPTION
**Pull request by fealho**
CU-86ayu71b2, Resolve #744

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86ayu71b2 RDT - Support Python 3.12 #744](https://app.clickup.com/t/86ayu71b2)

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
         os: [ubuntu-latest, macos-latest, windows-latest]
     steps:
     - uses: actions/checkout@v1
```
Comment (gsheni) [inline .github/workflows/integration.yml:16]:
```suggestion
    - uses: actions/checkout@v4
```

File: pyproject.toml [hunk 2]
```diff
@@ -12,24 +12,28 @@ classifiers = [
...
+    'Programming Language :: Python :: 3.12',
     'Topic :: Scientific/Engineering :: Artificial Intelligence',
 ]
 keywords = ['machine learning', 'synthetic data generation', 'benchmark', 'generative models']
 dynamic = ['version']
 license = { text = 'BSL-1.1' }
-requires-python = '>=3.8,<3.12'
+requires-python = '>=3.8,<3.13'
 readme = 'README.md'
 dependencies = [
```
Comment (gsheni) [inline pyproject.toml:23]: Seems like the Windows tests are failing. Perhaps you'll need to require `setuptools` for Windows?

**Review by pvk-developer — APPROVED** [review]
LGTM!

**Pull request merged**